lib/grinder_tools.dart

```diff
@@ -84,53 +85,57 @@ void runProcess(String executable,
       environment: environment);
 
   if (!quiet) {
-    if (result.stdout != null && !result.stdout.isEmpty) {
+    if (result.stdout != null && result.stdout.isNotEmpty) {
       log(result.stdout.trim());
     }
   }
 
-  if (result.stderr != null && !result.stderr.isEmpty) {
+  if (result.stderr != null && result.stderr.isNotEmpty) {
     log(result.stderr);
   }
 
   if (result.exitCode != 0) {
-    throw new GrinderException(
-        "${executable} failed with a return code of ${result.exitCode}");
+    throw new ProcessException(executable, result.exitCode, result.stderr);
   }
+
+  return result.stdout;
 }
 
-/**
- * Run the given executable, with optional arguments and working directory.
- */
-Future runProcessAsync(String executable,
+/// Run the given [executable], with optional [arguments] and [workingDirectory].
+///
+/// Returns a future for the stdout.
+Future<String> runProcessAsync(String executable,
     {List<String> arguments : const [],
      bool quiet: false,
      String workingDirectory}) {
-  log("${executable} ${arguments.join(' ')}");
+
+  if (!quiet) log("$executable ${arguments.join(' ')}");
+
+  List<int> stdout = [], stderr = [];
 
   return Process.start(executable, arguments, workingDirectory: workingDirectory)
       .then((Process process) {
+
     // Handle stdout.
-    process.stdout.listen((List<int> data) {
-      if (!quiet) {
-        String str = new String.fromCharCodes(data).trimRight();
-        if (str.isNotEmpty) log(str);
-      }
-    });
+    var broadcastStdout = process.stdout.asBroadcastStream();
+    var stdoutLines = _toLineStream(broadcastStdout);
```

Comment:
This way the line breaks in the grinder scripts output matces that of the run process.